Improve UI scaling for 4k displays

LordChunk opened the pull request:
Converted all pixel units to rem. Also added logic in main.css for scaling the root font size for higher resolution displays.

Fixes: https://github.com/overextended/ox_lib/issues/669

LordChunk committed:
Converted pixel units to rem and added font scaling for high res displays

LordChunk committed:
Removed comment

Signed-off-by: Job <LordChunk@users.noreply.github.com>

LordChunk committed:
Update CircleProgressbar.tsx

Signed-off-by: Job <LordChunk@users.noreply.github.com>

LukeWasTakenn commented:
Thank you so much for the PR!

I'll give it a proper review over the weekend.

LordChunk commented:
@LukeWasTakenn a question regarding the GitHub Actions pipelines. I tried running a build using the Node and pnpm versions you're currently using to build ox_lib. These currently fail because I updated the lockfile using a more up to date version of pnpm (v10).

I can make another PR to update the release scripting to Node 22 + pnpm 10, but that might mean you also need to upgrade the server version requirement to build `/server:12913` to use Node 22 server side. The other option is that I simply remove the lockfile changes from the PR.

LukeWasTakenn commented:
Let's avoid bumping the node version, for now at least. Though you should still be able to use pnpm 10, should build fine after it generates an updated lock file...?

LordChunk commented:
The latest version of pnpm that's compatible with Node v16 is pnpm v8. 
https://pnpm.io/installation#compatibility

I did some extra digging and it seems to be related to one of the patches which are run when installing dependencies:
```
[Release/Build and Create Tagged Release] ⭐ Run Main Install dependencies
[Release/Build and Create Tagged Release]   🐳  docker exec cmd=[bash -e /var/run/act/workflow/5] user= workdir=web
|  WARN  Ignoring not compatible lockfile at /workspaces/ox_lib/web/pnpm-lock.yaml
| Progress: resolved 1, reused 0, downloaded 0, added 0
|  WARN  deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
| Progress: resolved 36, reused 0, downloaded 30, added 0
| Progress: resolved 145, reused 0, downloaded 128, added 0
|  ERR_PNPM_PATCH_NOT_APPLIED  The following patches were not applied: react-hot-toast@2.4.0
| 
| Either remove them from "patchedDependencies" or update them to match packages in your dependencies.
[Release/Build and Create Tagged Release]   ❌  Failure - Main Install dependencies [3.524802643s]
[Release/Build and Create Tagged Release] exitcode '1': failure
```

I've also tried running this on the master branch and it throws the same error.

LordChunk commented:
@LukeWasTakenn did you manage to test this PR?

LukeWasTakenn commented:
no, the state of the project is up in the air right now.

LordChunk commented:
Ah okay 😅, I'll leave this as is for now. Should you need any more assistance with maintaining Overextended, feel free to reach out on Discord (lord_chunk)

antond15 committed:
chore: update organization name

antond15 committed:
refactor: use bun instead of pnpm

antond15 committed:
ci(create-release): restructure and update versions

github-actions[bot] committed:
chore: bump version to v3.30.7
